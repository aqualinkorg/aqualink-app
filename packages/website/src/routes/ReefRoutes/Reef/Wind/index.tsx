import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Card,
  CardContent,
  Typography,
  CardHeader,
  Grid,
} from "@material-ui/core";

import type { Data } from "../../../../store/Reefs/types";
import { sortDailyData } from "../../../../helpers/sortDailyData";
import { formatNumber } from "../../../../helpers/numberUtils";
import wind from "../../../../assets/wind.svg";
import clouds from "../../../../assets/clouds.svg";
import clouds2 from "../../../../assets/clouds2.svg";
import arrow from "../../../../assets/directioncircle.svg";

const Wind = ({ dailyData, classes }: WindProps) => {
  const sortByDate = sortDailyData(dailyData, "desc");
  const { maxWindSpeed, windDirection } = sortByDate[0];

  return (
    <Card className={classes.card}>
      <CardHeader
        className={classes.header}
        title={
          <Grid container justify="flex-start">
            <Grid item xs={2}>
              <img alt="wind" src={wind} />
            </Grid>
            <Grid item xs={8}>
              <Typography className={classes.cardTitle} variant="h6">
                WIND
              </Typography>
            </Grid>
          </Grid>
        }
      />
      <CardContent className={classes.contentWrapper}>
        <Grid
          style={{ height: "100%" }}
          container
          justify="center"
          item
          xs={12}
        >
          <Grid container justify="flex-end" item xs={9}>
            <img style={{ height: "5rem" }} alt="clouds" src={clouds} />
          </Grid>
          <Grid container justify="center" item xs={10}>
            <Grid item xs={6}>
              <Typography
                className={classes.contentTitles}
                color="textSecondary"
                variant="subtitle2"
              >
                SPEED
              </Typography>
              <Grid container alignItems="baseline">
                <Typography
                  className={classes.contentValues}
                  color="textSecondary"
                >
                  {formatNumber(maxWindSpeed, 1) || "- -"}
                </Typography>
                {maxWindSpeed && (
                  <Typography
                    className={classes.contentUnits}
                    color="textSecondary"
                    variant="h6"
                  >
                    KM/H
                  </Typography>
                )}
              </Grid>
            </Grid>
            <Grid item xs={6}>
              <Typography
                className={classes.contentTitles}
                color="textSecondary"
                variant="subtitle2"
              >
                DIRECTION
              </Typography>
              <Grid container alignItems="baseline">
                {windDirection && (
                  <img
                    style={{ transform: `rotate(${windDirection}deg)` }}
                    className={classes.arrow}
                    alt="arrow"
                    src={arrow}
                  />
                )}
                <Typography
                  className={classes.contentValues}
                  color="textSecondary"
                  variant="h3"
                >
                  {windDirection ? `${windDirection}\u00B0` : "- -"}
                </Typography>
              </Grid>
            </Grid>
          </Grid>
          <Grid
            style={{ height: "5rem" }}
            container
            justify="flex-start"
            item
            xs={7}
          >
            <img style={{ height: "100%" }} alt="clouds" src={clouds2} />
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    card: {
      height: "100%",
      width: "100%",
      backgroundColor: "#eff0f0",
      display: "flex",
      flexDirection: "column",
    },
    cardTitle: {
      lineHeight: 1.5,
      margin: "0 0 0.5rem 1rem",
      color: theme.palette.primary.main,
    },
    header: {
      flex: "0 1 auto",
      paddingLeft: "2rem",
      paddingBottom: 0,
    },
    contentWrapper: {
      flex: "1 1 auto",
      padding: 0,
    },
    content: {
      padding: "0 0 2rem 2rem",
    },
    contentTitles: {
      lineHeight: 1.33,
    },
    contentValues: {
      fontSize: 42,
      [theme.breakpoints.between("md", "lg")]: {
        fontSize: 32,
      },
      marginRight: "0.25rem",
      fontWeight: 300,
    },
    contentUnits: {
      [theme.breakpoints.between("md", "lg")]: {
        fontSize: 12,
      },
    },
    arrow: {
      width: 20,
      height: 20,
      [theme.breakpoints.between("md", "lg")]: {
        width: 14,
        height: 14,
      },
    },
  });

interface WindIncomingProps {
  dailyData: Data[];
}

type WindProps = WithStyles<typeof styles> & WindIncomingProps;

export default withStyles(styles)(Wind);
