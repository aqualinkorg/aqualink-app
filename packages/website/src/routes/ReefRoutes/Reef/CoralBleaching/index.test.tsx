import React from "react";
import { render } from "@testing-library/react";

import Wind from ".";

test("renders as expected", () => {
  const dailyData = [
    {
      id: 1,
      date: "2020-07-01T14:25:18.008Z",
      minBottomTemperature: 25,
      maxBottomTemperature: 27,
      avgBottomTemperature: 26,
      degreeHeatingDays: 29,
      surfaceTemperature: 36,
      satelliteTemperature: 20,
      minWaveHeight: 1,
      maxWaveHeight: 3,
      avgWaveHeight: 2,
      waveDirection: 136,
      wavePeriod: 15,
      minWindSpeed: 3,
      maxWindSpeed: 5,
      avgWindSpeed: 4,
      windDirection: 96,
    },
  ];
  const { container } = render(<Wind dailyData={dailyData} />);
  expect(container).toMatchSnapshot();
});
