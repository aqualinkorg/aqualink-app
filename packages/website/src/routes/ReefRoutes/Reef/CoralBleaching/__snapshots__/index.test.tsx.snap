// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-card
    classname="Bleaching-card-1"
  >
    <div
      class="MuiCardHeader-root Bleaching-header-3"
    >
      <div
        class="MuiCardHeader-content"
      >
        <mock-typography
          classname="MuiCardHeader-title"
          component="span"
          display="block"
          variant="h5"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <mock-typography
                color="textSecondary"
                variant="h6"
              >
                HEAT STRESS ALERT LEVEL
              </mock-typography>
            </div>
          </div>
        </mock-typography>
      </div>
    </div>
    <mock-cardcontent
      classname="Bleaching-contentWrapper-8"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
        style="height: 100%;"
      >
        <img
          alt="alert-level"
          src="alert_lvl1.svg"
        />
      </div>
    </mock-cardcontent>
  </mock-card>
</div>
`;
